bug: ci fails with SyntaxError on bposd package

### Bug Description

CI started to fail when running the `pytest` target of `labeled_tests.yml` on `deltakit-explorer`, only with Python 3.10 and 3.11 due to the [`bposd`](https://pypi.org/project/bposd/) package including some invalid escape sequences `\i`.

### Minimal Reproducible Example (MRE)

See https://github.com/Deltakit/deltakit/actions/runs/19040550598/job/54447222124

The issue is repeatable (i.e., starting again the action does not solve it).

### Actual behavior

At tests collection, the following error is raised:

```
_ ERROR collecting deltakit-explorer/tests/circuit_generation/test_circuit_generation.py _
../.pixi/envs/deltakit-explorer-tests-py310/lib/python3.10/site-packages/_pytest/python.py:498: in importtestmodule
    mod = import_path(
../.pixi/envs/deltakit-explorer-tests-py310/lib/python3.10/site-packages/_pytest/pathlib.py:587: in import_path
    importlib.import_module(module_name)
../.pixi/envs/deltakit-explorer-tests-py310/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../.pixi/envs/deltakit-explorer-tests-py310/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
    exec(co, module.__dict__)
tests/circuit_generation/test_circuit_generation.py:4: in <module>
    from deltakit_explorer.codes._bivariate_bicycle_code import \
src/deltakit_explorer/codes/__init__.py:2: in <module>
    from deltakit_explorer.codes._bivariate_bicycle_code import (
src/deltakit_explorer/codes/_bivariate_bicycle_code.py:21: in <module>
    from deltakit_explorer.codes._css._css_code import CSSCode
src/deltakit_explorer/codes/_css/_css_code.py:23: in <module>
    from deltakit_explorer.codes._logicals import (
src/deltakit_explorer/codes/_logicals.py:22: in <module>
    from bposd.css import css_code  # type: ignore
E     File "/home/runner/work/deltakit/deltakit/.pixi/envs/deltakit-explorer-tests-py310/lib/python3.10/site-packages/bposd/css.py", line 162
E       print(" -lx \in ker{hz} AND lz \in ker{hx}: Fail")
E             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   SyntaxError: invalid escape sequence '\i'
```

### Expected behavior

No failure due to external packages.

### Additional context

The same test was not failing at least 4 days ago. The last run I found that did not fail I found is https://github.com/Deltakit/deltakit/actions/runs/18973856824/job/54188062088.

From [Python 3.12 changelog](https://docs.python.org/3/whatsnew/3.12.html):

> A backslash-character pair that is not a valid escape sequence now generates a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning), instead of [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning). For example, re.compile("\d+\.\d+") now emits a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) ("\d" is an invalid escape sequence, use raw strings for regular expression: re.compile(r"\d+\.\d+")). In a future Python version, [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError) will eventually be raised, instead of [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning). (Contributed by Victor Stinner in [gh-98401](https://github.com/python/cpython/issues/98401).)

I was not able to find a change between a successful run and a failed one.

The Python versions did not change (3.10.19 and 3.11.14), the `bposd` package last release was months ago.

~~I am not able to replicate that bug locally, `pixi run -e deltakit-explorer-tests-py310 tests` and `pixi run -e deltakit-explorer-tests-py311 tests` both succeed. The Python versions pixi uses locally are 3.10.18 and 3.11.13 (so one patch version behind) but I looked at the release notes for both and could not find anything that could cause this issue.~~

I am now able to replicate that bug locally, will investigate more later.

--- a/deltakit-explorer/pyproject.toml
+++ b/deltakit-explorer/pyproject.toml
@@ -41,7 +41,7 @@ dependencies = [
     "requests_toolbelt >=1.0.0",  # needed by gql, which doesn't automatically install it
     "pymatching >= 2.2.2",
     "galois >= 0.4.6",
-    "bposd >= 2.1",
+    "bposd @ git+https://github.com/quantumgizmos/bp_osd.git@8894ec654b24ae875c07e5a361dcae9a77d748ce",
     "pandas >= 2.0",
     "pandas-stubs",
     "types-seaborn",

